File: src/main/java/mods/eln/sim/ThermalResistor.java
package mods.eln.sim;

public class ThermalResistor implements IProcess {

    ThermalLoad a;
    ThermalLoad b;

    protected double thermalResistance;
    protected double thermalResistanceInverse;

    public ThermalResistor(ThermalLoad a, ThermalLoad b) {
        this.a = a;
        this.b = b;
        highImpedance();
    }

    @Override
    public void process(double time) {
        double power = (a.temperatureCelsius - b.temperatureCelsius) * thermalResistanceInverse;
        a.PcTemp -= power;
        b.PcTemp += power;
    }

    public double getPower() {
        return (a.temperatureCelsius - b.temperatureCelsius) * thermalResistanceInverse;
    }

    public void setThermalResistance(double thermalResistance) {
        this.thermalResistance = thermalResistance;
        thermalResistanceInverse = 1 / thermalResistance;
    }

    public double getThermalResistance() {
        return thermalResistance;
    }

    public void highImpedance() {
        setThermalResistance(1000000000.0);
    }
}

File: src/main/java/mods/eln/sim/ThermalResistor.kt
package mods.eln.sim

class ThermalResistor
//double P = 0;

(internal var a: ThermalLoad, internal var b: ThermalLoad) : IProcess {

    var r: Double = 0.0
        get() = field
        set(r) {
            field = r
            Rinv = 1 / r
        }
    protected var Rinv: Double = 1 / r

    val p: Double
        get() = (a.Tc - b.Tc) * Rinv

    init {
        highImpedance()
    }

    override fun process(time: Double) {
        val P = (a.Tc - b.Tc) * Rinv
        a.PcTemp -= P
        b.PcTemp += P
    }
    /*
	public double getVoltage() {
		return P * R;
	}*/

    fun highImpedance() {
        r = 1000000000.0
    }
}
